app/Http/Requests/CreateProjectRequest.php

```diff
@@ -0,0 +1,76 @@
+<?php
+
+namespace App\Http\Requests;
+
+use App\Project;
+
+class CreateProjectRequest extends Request
+{
+    /**
+     * Determine if the user is authorized to make this request.
+     *
+     * @return bool
+     */
+    public function authorize()
+    {
+        return $this->user() != null;
+    }
+
+    /**
+     * Get the validation rules that apply to the request.
+     *
+     * @return array
+     */
+    public function rules()
+    {
+        return [
+            'title' => 'required|unique:projects|' . $this->minMaxFormatter('title'),
+            'description' => 'required|' . $this->minMaxFormatter('description'),
+            'project-body' => 'required|' . $this->minMaxFormatter('project-body'),
+            'thumbnail' => 'required|image|max:2048'    // max image size 2mb
```

Comment:
Why create a separate function for setting max and min? Laravel has min:... and max:.. already.